evm/src/witness/memory.rs

```diff
@@ -202,12 +202,20 @@ impl Default for MemoryState {
     }
 }
 
-#[derive(Clone, Default, Debug)]
+#[derive(Clone, Debug)]
 pub(crate) struct MemoryContextState {
     /// The content of each memory segment.
     pub(crate) segments: [MemorySegmentState; Segment::COUNT],
 }
 
+impl Default for MemoryContextState {
+    fn default() -> Self {
+        Self {
+            segments: std::array::from_fn(|_| MemorySegmentState::default()),
+        }
+    }
+}
```

Comment:
Nit, but couldn't we do instead? I believe `MemorySegmentState` implements `Clone`.
```suggestion
impl Default for MemoryContextState {
    fn default() -> Self {
        Self {
            segments: [MemorySegmentState::default(); Segment::COUNT]
        }
    }
}
```

Reply:
Ah right, I forgot about this quirk. Then both the current version or Robin's suggestion look good to me.

Reply:
With Robin's suggestion, it is still complaining unfortunately, because there is a `Vec` in `MemorySegmentState`, which does not implement `Copy`. So I left the current version as is.